debian-hostname: init at 3.23

posch opened the pull request:
The Debian version of hostname is widely used on other distros. It offers a few more options (-I, -A).

It may be useful when writing scripts that need to be portable across distros.

###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

posch committed:
hostname-debian: init at 3.23

The Debian version of hostname is widely used on other distros.
It offers a few more options (-I, -A).

wentasah commented on pkgs/tools/networking/debian-hostname/default.nix:
```diff
@@ -0,0 +1,38 @@
+{ stdenv, lib, fetchurl }:
+
+stdenv.mkDerivation rec {
+  pname = "debian-hostname";
```
Personally, I'd call the package `hostname-debian`. Hostname is more important than Debian so it should come first.

wentasah commented on pkgs/tools/networking/debian-hostname/default.nix:
```diff
@@ -0,0 +1,38 @@
+  doCheck = false;
+  installFlags = [];
+  installPhase = ''
+    sed -i.orig '/install/ s,-[og] root,,g' Makefile
```
No need to backup the original Makefile. Maybe, the following would be more readable:
```sh
substituteInPlace Makefile --replace 'install -o root -g root' 'install'
```

SuperSandro2000 commented on pkgs/tools/networking/debian-hostname/default.nix:
```diff
@@ -0,0 +1,38 @@
+  '';
+
+  meta = with lib; {
+    description = "Collection of common network programs";
+
+    longDescription = ''
+      utility to set/show the host name or domain name
+    '';
+
+    homepage = "https://tracker.debian.org/pkg/hostname";
+    license = licenses.gpl2Plus;
+
+    maintainers = with maintainers; [ ];
+    platforms = platforms.gnu;
```
```suggestion
    description = "Collection of network utilities to set/show the host name or domain name";
    homepage = "https://tracker.debian.org/pkg/hostname";
    license = licenses.gpl2Plus;
    maintainers = with maintainers; [ ];
    platforms = platforms.gnu;
```
meta.maintainers must be set.

SuperSandro2000 commented on pkgs/tools/networking/debian-hostname/default.nix:
```diff
@@ -0,0 +1,38 @@
+  buildInputs = [];
+  doCheck = false;
+  installFlags = [];
+  installPhase = ''
+    sed -i.orig '/install/ s,-[og] root,,g' Makefile
+    make install BINDIR=$out/bin MANDIR=$out/share/man
+  '';
```
Please try to use nix' default installPhase.

SuperSandro2000 commented on pkgs/tools/networking/debian-hostname/default.nix:
```diff
@@ -0,0 +1,38 @@
+
+  outputs = ["out"];
+
+  patches = [];
+
+  nativeBuildInputs = [];
+  buildInputs = [];
+  doCheck = false;
```
```suggestion
  doCheck = false;
```

Please add a comment why tests are disabled.

SuperSandro2000 commented on pkgs/tools/networking/debian-hostname/default.nix:
```diff
@@ -0,0 +1,38 @@
+    url = "http://deb.debian.org/debian/pool/main/h/hostname/hostname_${version}.tar.gz";
+    sha256 = "sha256-vG0ZVLIoSYaf+LKmAuOfCLFwL2htS1jdeSfN61tIdu8=";
+  };
+
+  outputs = ["out"];
```
```suggestion
```
that is the default, no need to repeat it

SuperSandro2000 commented on pkgs/tools/networking/debian-hostname/default.nix:
```diff
@@ -0,0 +1,38 @@
+  version = "3.23";
+
+  src = fetchurl {
+    url = "http://deb.debian.org/debian/pool/main/h/hostname/hostname_${version}.tar.gz";
```
```suggestion
    url = "https://deb.debian.org/debian/pool/main/h/hostname/hostname_${version}.tar.gz";
```

wentasah reviewed (changes requested):
This would be definitely a useful addition. I had to fix incompatible hostname flags several times in the past. Thank you.

Below, I suggest few changes.

posch committed:
maintainers: add posch

wentasah reviewed (approved):
LGTM

posch commented:
Thanks for the reviews. All your points make sense, and seem to improve consistency with how other packages were written. I think I've addressed them all. I also renamed the package to hostname-debian.

SuperSandro2000 commented on pkgs/tools/networking/hostname-debian/default.nix:
```diff
@@ -0,0 +1,28 @@
+    sha256 = "sha256-vG0ZVLIoSYaf+LKmAuOfCLFwL2htS1jdeSfN61tIdu8=";
+  };
+
+  prePatch = ''
```
```suggestion
  postPatch = ''
```

posch replied:
Can you explain your preference for postPatch?

grep -rC4 -- '-o root' pkgs
shows lots of places with this type of patch. Some use a separate patch file, otherwise it's:
- prePatch (e.g. super, pptp, x2goserver)
- postPatch (datefudge, ipad_charge, pptpd)
- postConfigure (pmount, bsdgames)
- preBuild (crunch, dd_rescue, cron)
- patchPhase (gt5, pal, pam_ccreds)

I can't see a clear winner.

My thought was that prePatch will apply to unmodified sources, while postPatch will have to deal with sources that may have been modified. But since patchPhase is a nop here, prePatch and postPatch are equivalent.

SuperSandro2000 replied:
If there is a patch that touches the surrounding lines then it wouldn't apply any longer and it is not my personal preference but a general rule.

> patchPhase (gt5, pal, pam_ccreds)

That is completely wrong and breaks patches.

> I can't see a clear winner.

You're methodology is not very good since it also accounts for old packages that are not up to date with current standards.

posch commented:
I'm fine with postPatch and "If there is a patch that touches the surrounding lines then it wouldn't apply any longer" is a good argument.

posch commented:
@SuperSandro2000 Is there anything I missed? What keeps your from merging?

SuperSandro2000 commented:
I just found time to look at this again.
